File: src/main/java/org/frawa/elmtest/run/ElmTestRunConfigurationType.java
package org.frawa.elmtest.run;

import com.intellij.execution.configurations.ConfigurationFactory;
import com.intellij.execution.configurations.ConfigurationType;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

import static org.frawa.elmtest.run.ElmTestConfigurationFactory.RUN_ICON;

public class ElmTestRunConfigurationType implements ConfigurationType {
    @Override
    public String getDisplayName() {
        return "Elm Test";
    }

    @Override
    public String getConfigurationTypeDescription() {
        return "Elm Test Runner";
    }

    @Override
    public Icon getIcon() {
        return RUN_ICON;
    }

    @NotNull
    @Override
    public String getId() {
        return "ELM_TEST_RUN_CONFIGURATION";
    }

    @Override
    public ConfigurationFactory[] getConfigurationFactories() {
        return new ConfigurationFactory[]{new ElmTestConfigurationFactory(this)};
    }


}

File: src/main/kotlin/org/elm/ide/test/run/ElmTestRunConfigurationType.kt
package org.elm.ide.test.run

import com.intellij.execution.configurations.ConfigurationType
import org.elm.ide.test.run.ElmTestConfigurationFactory.Companion.RUN_ICON

class ElmTestRunConfigurationType : ConfigurationType {

    override fun getDisplayName() = "Elm Test"

    override fun getConfigurationTypeDescription() = "Elm Test Runner"

    override fun getIcon() = RUN_ICON

    override fun getId() = "ELM_TEST_RUN_CONFIGURATION"

    override fun getConfigurationFactories() = arrayOf(ElmTestConfigurationFactory(this))

}
